### PR TITLE
feat: add allEqual

### PR DESCRIPTION
**Pull request by char0n**
Continuation of https://github.com/char0n/ramda-adjunct/pull/513

### CODE REVIEW

File: test/allEqual.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { assert } from 'chai';
+
+import * as RA from '../src/index';
+
```
Comment (Undistraction) [inline test/allEqual.js]: Might be worth adding a test for an array of `[1, 1, '1']` or is that actually testing `equal`?

File: test/allEqual.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  context('when items are not equal', function() {
+    specify('should return false', function() {
+      assert.isFalse(RA.allEqual([1, 1, 2, 1, 1]));
+    });
+  });
+
+  context('when empty list provided', function() {
+    specify('should return true', function() {
+      assert.isTrue(RA.allEqual([]));
+    });
```
Comment (Undistraction) [inline test/allEqual.js]: What happens if no value is passed: `allEqual()`?
What happens if the same function is passed multiple times?
```
const f = () => {}
RA.allEqual([f, f, f]))
```

File: src/allEqual.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { pipe, uniq } from 'ramda';
+
+import lengthLte from './lengthLte';
+
+// Original idea for this function was conceived by https://github.com/jackmellis
+// in https://github.com/char0n/ramda-adjunct/pull/513.
+
+/**
+ * Checks that all elements in a list are equal.
```
Comment (Undistraction) [inline src/allEqual.js]: I think you need to note what 'equal' means in this context i.e. 'Checks that all elements in a list are equal using `R. uniq` which uses `R.equals` for comparison'.

File: test/allEqual.js (hunk 1 above)
Comment (char0n) [reply test/allEqual.js]: Yeah it's testing equal but we can have explicit test for this to demonstrate usecase for different types. I'll add it.

File: test/allEqual.js (hunk 2 above)
Comment (char0n) [reply test/allEqual.js]: Can add test for testing functions. 

> What happens if no value is passed: allEqual()?
I think we don't care. The signature is clear and everything outside of the signature
is unspecified.

File: src/allEqual.js (hunk 3 above)
Comment (char0n) [reply src/allEqual.js]: Will explain it explicitly.

**Comment by char0n** [comment]
@rockymadden will investigate the algorithm, thx

**Comment by char0n** [comment]
All CRC addressed. Requesting for re-review ;]

**Comment by Undistraction** [comment]
Changes look good, but failing in CI.

**Comment by rockymadden** [comment]
Can’t write our code now, away from laptop. The idea, however, is this: If you have an array of a million items and you use uniq on it, how many times will have equals have been run? Now if you use reduceWhile where you short circuit on the first non equality how many? The worst case scenario in the latter is the best case in the former right?

**Comment by char0n** [comment]
The CI failured was caused by https://github.com/ramda/ramda/pull/1992; trying to compensate

**Comment by char0n** [comment]
I'll go for 2.9.0 release. We have some new keywords in package.json that needs to go out ASAP.

**Pull request merged**